StripeCore/StripeCore/Source/DownloadManager/DownloadManager.swift

```diff
@@ -53,177 +41,114 @@ import UIKit
-                CachedURLResponse(response: response, data: data),
-                for: urlRequest
-            )
-            updateHandler(image)
-            blockingDownloadSemaphore.signal()
-        }
-        task.resume()
-        blockingDownloadSemaphore.wait()
-        return blockingDownloadedImage ?? placeholder
+        // Immediately return a placeholder, when the download operation completes `updateHandler` will be called with the downloaded image
+        return placeholder
     }
 
-    func downloadImageAsync(url: URL, placeholder: UIImage?, updateHandler: UpdateImageHandler?) -> UIImage {
-        let placeholder = placeholder ?? imagePlaceHolder()
-        let imageName = imageNameFromURL(url: url)
-        if let image = cachedImageNamed(imageName) {
+    // Common download function
+    private func downloadImage(url: URL, placeholder: UIImage) async -> UIImage {
+        let imageName = url.lastPathComponent
+        // Early exit for cached images
+        if let image = await imageCache.cachedImageNamed(imageName) {
             return image
         }
-        let urlRequest = URLRequest(url: url, cachePolicy: .useProtocolCachePolicy)
-        let task = self.session.downloadTask(with: url) { [weak self] tempURL, response, _ in
-            guard let self = self,
-                  let tempURL = tempURL,
-                let response = response,
-                let data = self.getDataFromURL(tempURL),
-                let image = self.persistToMemory(data, forImageName: imageName)
-            else {
-                self?.pendingRequestsSemaphore.wait()
-                self?.pendingRequests.removeValue(forKey: imageName)
-                self?.pendingRequestsSemaphore.signal()
-                return
-            }
-            self.urlCache?.storeCachedResponse(
-                CachedURLResponse(response: response, data: data),
-                for: urlRequest
-            )
-
-            self.pendingRequestsSemaphore.wait()
-            self.pendingRequests.removeValue(forKey: imageName)
-            let updates = self.updateHandlers[imageName] ?? []
-            self.updateHandlers.removeValue(forKey: imageName)
-            self.pendingRequestsSemaphore.signal()
-
-            for updateHandler in updates {
-                updateHandler(image)
-            }
-        }
 
-        self.pendingRequestsSemaphore.wait()
-        guard self.pendingRequests[imageName] == nil else {
-            addUpdateHandlerWithoutLocking(updateHandler, forImageName: imageName)
-            self.pendingRequestsSemaphore.signal()
+        let urlRequest = URLRequest(url: url, cachePolicy: .useProtocolCachePolicy)
+        do {
+            let (data, response) = try await session.data(for: urlRequest)
+            let image = try persistToMemory(data, forImageName: imageName) // Throws a Error.failedToMakeImageFromData
+            diskCache?.storeCachedResponse(CachedURLResponse(response: response, data: data), for: urlRequest)
```

Comment:
Why do we manually handle the cache?  And if we do, why set it on urlSession.configuration?  Seems like it should be one or the other.